servers/src/main/java/tachyon/master/block/journal/BlockInfoEntry.java

```diff
@@ -50,4 +51,23 @@ public long getLength() {
   public JournalEntryType getType() {
     return JournalEntryType.BLOCK_INFO;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
```

Comment:
Arguably the code runs faster if we are comparing object to itself.
But if that scenario is considered rare (compare to one self),
then we should take it out, to make the code more compact.
Just let me know. I am fine either way...

Thanks,
Yu

On Fri, Oct 30, 2015 at 10:35 AM, Calvin Jia <notifications@github.com>
wrote:

> In servers/src/main/java/tachyon/master/block/journal/BlockInfoEntry.java
> <https://github.com/amplab/tachyon/pull/1939#discussion_r43530041>:
>
> > @@ -50,4 +51,23 @@ public long getLength() {
> >    public JournalEntryType getType() {
> >      return JournalEntryType.BLOCK_INFO;
> >    }
> > +
> > +  @Override
> > +  public boolean equals(Object o) {
> > +    if (this == o) {
> > +      return true;
> > +    }
>
> Is there any point in this check (is the following check a superset of
> this one)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amplab/tachyon/pull/1939/files#r43530041>.
>